lib/vsc/administration/slurm/sync.py

```diff
@@ -300,18 +457,92 @@ def slurm_institute_accounts(slurm_account_info, clusters, host_institute, insti
     return commands
 
 
+def get_cluster_accounts(slurm_account_info, cluster):
+    return dict([
+            (acct.Account, int(acct.Share))
+            for acct in slurm_account_info
+            if acct and acct.Cluster == cluster
+        ])
+
+
+def get_cluster_qos(slurm_qos_info, cluster):
+    """Returns a list of QOS names related to the given cluster"""
+
+    return [qi.Name for qi in slurm_qos_info if qi.Name.startswith(cluster)]
+
+
+def slurm_project_qos(projects, slurm_qos_info, clusters):
+    """Check for new/changed projects and set their QOS accordingly"""
+    commands = []
+    for cluster in clusters:
+        cluster_qos_names = set(get_cluster_qos(slurm_qos_info, cluster))
+        project_qos_names = set([
+            "{cluster}-{project_name}".format(cluster=cluster, project_name=p.name) for p in projects
+        ])
+
+        for project in projects:
+            qos_name = "{0}-{1}".format(cluster, project.name)
+            if qos_name not in cluster_qos_names:
+                commands.append(create_add_qos_command(qos_name))
+                commands.append(create_modify_qos_command(qos_name, {
+                    "GRPTRESMins": "cpu={cpuminutes},gpu={gpuminutes}".format(
+                        cpuminutes=60*int(project.cpu_hours)
```

Comment:
teh cpu_hours and gpu_hours are already int

Reply:
they are now, they were not before. this is not issue and it is far safer, just in case. There is _no_ type checking, remember.